### PR TITLE
Add a docker compatibility demo

### PR DESCRIPTION
**Pull request by ipbabble**
Signed-off-by: William Henry @ipbabble <whenry@redhat.com>

### CODE REVIEW

File: demos/docker-compatibility-demo.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+buildah config --created-by "${quayuser}"  $newcontainer
+buildah config --author "${myname} @${quayuser}" --label name=$demoimg $newcontainer
+read -p "${green}Inspect the container image meta data${yellow}"
+buildah inspect $newcontainer
+read -p "${green}Commit the container to an OCI image called ${demoimg}.${reset}"
+buildah commit $newcontainer $demoimg
+read -p "${green}List the images we have.${reset}"
+buildah images
+read -p "${blue}Run the container using Podman. Check ${bold}http://localhost${reset}"
+containernum=$(podman run -d -p 80:80 $demoimg)
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: Might be a @mheon question, but I'm getting:

```
+ read -p 'Run the container using Podman. Check http://localhost'
Run the container using Podman. Check http://localhost
++ podman run -d -p 80:80 dockercompatibilitydemo
ERRO[0000] Error removing store for partially-created runtime: A layer is mounted: layer is in use by a container 
+ containernum=4da626b767ebc797a97b850ca877f4a3f5ec90b75fac93f9eb2bbce031fbc2d7
+ read -p '}Check that nginx is up and running with our new page'
```

File: demos/docker-compatibility-demo.sh [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+read -p "$cyan}Check that nginx is up and running with our new page${reset}"
+curl localhost
+read -p "${blue}Stop the container and rm it${reset}"
+podman ps                                                                                   
+podman stop $containernum                                                                            
+podman rm $containernum                                                                              
+read -p "${cyan}Check that nginx is down${reset}"
+curl localhost
+read -p "${cyan}Start the Docker daemon${reset}"
+systemctl start docker
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: might want to change this to 'restart docker'  That will start it if it's down or restart it if it's up.

File: demos/docker-compatibility-demo.sh [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+buildah config --author "${myname} @${quayuser}" --label name=$demoimg $newcontainer
+read -p "${green}Inspect the container image meta data${yellow}"
+buildah inspect $newcontainer
+read -p "${green}Commit the container to an OCI image called ${demoimg}.${reset}"
+buildah commit $newcontainer $demoimg
+read -p "${green}List the images we have.${reset}"
+buildah images
+read -p "${blue}Run the container using Podman. Check ${bold}http://localhost${reset}"
+containernum=$(podman run -d -p 80:80 $demoimg) 
+read -p "$cyan}Check that nginx is up and running with our new page${reset}"
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: Maybe most people will know, but would it be good to tell them to do 'curl localhost:80' to do the check?

File: demos/docker-compatibility-demo.sh [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+buildah run $newcontainer -- ${pkgmgr} -y install nginx && ${pkgmgr} -y clean all 
+read -p "${green}Make some nginx config and home page changes ${reset}"
+buildah run $newcontainer bash -c 'echo "daemon off;" >> /etc/nginx/nginx.conf'
+buildah run $newcontainer bash -c 'echo "nginx on OCI Fedora image, built using Buildah" > /usr/share/nginx/html/index.html'
+read -p "${green}Use buildah config to expose the port and set the entrypoint${reset}"
+buildah config --port 80 --entrypoint /usr/sbin/nginx $newcontainer
+read -p "${green}Set other meta data using buildah config${reset}"
+buildah config --created-by "${quayuser}"  $newcontainer
+buildah config --author "${myname} @${quayuser}" --label name=$demoimg $newcontainer
+read -p "${green}Inspect the container image meta data${yellow}"
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: probably doesn't matter, but I think you want {reset} and not {yellow} at the end of the line.

File: demos/docker-compatibility-demo.sh [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+systemctl restart docker
+read -p "${red}List the Docker images in the repository - should be empty${reset}"
+docker images                                                                          
+read -p "${blue}Push the image to the local Docker repository using docker-daemon${reset}"
+podman push $demoimg docker-daemon:$quayuser/dockercompatibilitydemo:latest       
+read -p "${red}List the Docker images in the repository${reset}"
+docker images                                                                          
+read -p "${red}Start the container from the new Docker repo image${reset}"
+dockercontainer=$(docker run -d -p 80:80 $quayuser/$demoimg)
+read -p "$cyan}Check that nginx is up and running with our new page${reset}"
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: "$cyan" to "${cyan"

File: demos/docker-compatibility-demo.sh [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+curl localhost
+read -p "${red}Stop the container and rm it${reset}"
+docker rm $dockercontainer                                                                            
+docker rm $dockercontainer
+read -p "$cyan}Stop Docker${reset}"
```
Comment (TomSweeneyRedHat) [inline demos/docker-compatibility-demo.sh]: "cyan}" to "{cyan}" here too.

File: demos/docker-compatibility-demo.sh (hunk 4 above)
Comment (ipbabble) [reply demos/docker-compatibility-demo.sh]: Nope. Intentional

File: demos/docker-compatibility-demo.sh (hunk 5 above)
Comment (ipbabble) [reply demos/docker-compatibility-demo.sh]: I fixed this. Ugh

File: demos/docker-compatibility-demo.sh (hunk 6 above)
Comment (ipbabble) [reply demos/docker-compatibility-demo.sh]: What happened all these?

Comment (ipbabble) [reply demos/docker-compatibility-demo.sh]: I’ll fix when I get back. Argh!!!!!!!

**Comment by TomSweeneyRedHat** [comment]
LGTM.  Thanks @ipbabble !

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**